aeneas/src/ir/VariantSolver.v3

```diff
@@ -0,0 +1,330 @@
+// Norm for variants (both superclass and case class)
+class VariantNorm extends TypeNorm {
```

Comment:
There are a lot of different classes now, and it looks like `VariantNorm` and `VariantParentNorm` are basically just markers (i.e. common superclasses) for the several subclasses below. I generally try to avoid that. Can we reduce the number of classes and maybe use a field to distinguish between an unboxed data norm and an unboxed case norm? (i.e. the only different right now is that the case norm has a tag).